src/darsia/measure/wasserstein.py

```diff
@@ -2090,6 +2117,686 @@ def _solve(self, flat_mass_diff: np.ndarray) -> tuple[float, np.ndarray, dict]:
         return new_distance, solution_i, info
 
 
+class WassersteinDistanceGproxPGHD(darsia.EMD):
```

Comment:
I did not go into the details and trust you that the method is doing what is intended, seeing you added all required tests.